src/arch/xtensa/include/arch/atomic.h

```diff
@@ -11,6 +11,9 @@
 #define __ARCH_ATOMIC_H__
 
 #include <stdint.h>
+#ifdef XCHAL_HAVE_EXCLUSIVE
+#include <xtensa/tie/xt_core.h>
```

Comment:
So these are not available with GCC build. 
```
                 from /home/sof/work/sof.git/src/platform/tigerlake/lib/clk.c:9:
/home/sof/work/sof.git/src/arch/xtensa/include/arch/atomic.h:15:10: fatal error: xtensa/tie/xt_core.h: No such file or directory
   15 | #include <xtensa/tie/xt_core.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
```
Can we make this a Kconfig option i.e. 

```#if XCHAL_HAVE_EXCLUSIVE && CONFIG_XTENSA_EXCLUSIVE && __XCC__``` means we only build this code if it's selected in Kconfig and we are using Cadence compiler (GCC would still build).